### ISSUE TITLE
jetty timeout ?  o.e.j.i.ChannelEndPoint  java.io.IOException: Connection reset by peer

### ISSUE THREAD

**Issue by lucilecoutouly** [spotify/heroic#333]
```powershell 
2017-09-15 07:50:41,024 DEBUG o.e.j.u.t.QueuedThreadPool [qtp681747431-167] run EPC Pend/org.eclipse.jetty.io.ManagedSelector$SelectorProducer@2c361e6a 
2017-09-15 07:50:41,024 DEBUG o.e.j.u.t.s.ExecuteProduceConsume [qtp681747431-167] EPC Pend/org.eclipse.jetty.io.ManagedSelector$SelectorProducer@2c361e6a run 
2017-09-15 07:50:41,024 DEBUG o.e.j.u.t.s.ExecuteProduceConsume [qtp681747431-167] EPC Prod/org.eclipse.jetty.io.ManagedSelector$SelectorProducer@2c361e6a produce enter 
2017-09-15 07:50:41,024 DEBUG o.e.j.u.t.s.ExecutingExecutionStrategy [qtp681747431-112] EPC Pend/org.eclipse.jetty.io.ManagedSelector$SelectorProducer@2c361e6a invoke  CEP:SocketChannelEndPoint@63ccefa7{/10.42.136.138:52600<->/10.42.157.16:8080,OSHUT,fill=FI,flush=-,to=2/30000}{io=0/0,kio=0,kro=1}->HttpConnection@7d66662[p=HttpParser{s=CLOSE,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null}:runFillable:BLOCKING 
2017-09-15 07:50:41,024 DEBUG o.e.j.u.t.s.ExecuteProduceConsume [qtp681747431-167] EPC Prod/org.eclipse.jetty.io.ManagedSelector$SelectorProducer@2c361e6a producing 
2017-09-15 07:50:41,024 DEBUG o.e.j.i.FillInterest [qtp681747431-112] FillInterest@5765eac6{true,AC.ReadCB@7d66662{HttpConnection@7d66662[p=HttpParser{s=CLOSE,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null}<-SocketChannelEndPoint@63ccefa7{/10.42.136.138:52600<->/10.42.157.16:8080,OSHUT,fill=FI,flush=-,to=2/30000}{io=0/0,kio=0,kro=1}->HttpConnection@7d66662[p=HttpParser{s=CLOSE,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null}}} fillable AC.ReadCB@7d66662{HttpConnection@7d66662[p=HttpParser{s=CLOSE,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null}<-SocketChannelEndPoint@63ccefa7{/10.42.136.138:52600<->/10.42.157.16:8080,OSHUT,fill=FI,flush=-,to=2/30000}{io=0/0,kio=0,kro=1}->HttpConnection@7d66662[p=HttpParser{s=CLOSE,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null}} 
2017-09-15 07:50:41,024 DEBUG o.e.j.s.HttpConnection [qtp681747431-112] HttpConnection@7d66662[p=HttpParser{s=CLOSE,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null}<-SocketChannelEndPoint@63ccefa7{/10.42.136.138:52600<->/10.42.157.16:8080,OSHUT,fill=-,flush=-,to=2/30000}{io=0/0,kio=0,kro=1}->HttpConnection@7d66662[p=HttpParser{s=CLOSE,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null} onFillable enter HttpChannelState@5265af91{s=IDLE a=NOT_ASYNC i=true r=!P!U w=false} null 
2017-09-15 07:50:41,025 DEBUG o.e.j.i.ChannelEndPoint [qtp681747431-112]  java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at org.eclipse.jetty.io.ChannelEndPoint.fill(ChannelEndPoint.java:240)
	at org.eclipse.jetty.server.HttpConnection.fillRequestBuffer(HttpConnection.java:322)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:231)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:112)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124)
	at org.eclipse.jetty.util.thread.Invocable.invokePreferred(Invocable.java:122)
	at org.eclipse.jetty.util.thread.strategy.ExecutingExecutionStrategy.invoke(ExecutingExecutionStrategy.java:58)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:201)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:133)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:672)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:590)
	at java.lang.Thread.run(Thread.java:748)

2017-09-15 07:50:41,026 DEBUG o.e.j.i.ChannelEndPoint [qtp681747431-112] doClose SocketChannelEndPoint@63ccefa7{/10.42.136.138:52600<->/10.42.157.16:8080,CLOSED,fill=-,flush=-,to=4/30000}{io=0/0,kio=0,kro=1}->HttpConnection@7d66662[p=HttpParser{s=CLOSE,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null} 
2017-09-15 07:50:41,027 DEBUG o.e.j.i.WriteFlusher [qtp681747431-112] ignored: WriteFlusher@6f3dbcb5{IDLE}->null {} 
2017-09-15 07:50:41,027 DEBUG o.e.j.i.FillInterest [qtp681747431-112] FillInterest@5765eac6{false,null} onClose null 
2017-09-15 07:50:41,027 DEBUG o.e.j.i.ManagedSelector [qtp681747431-112] Queued change org.eclipse.jetty.io.ManagedSelector$$Lambda$1418/652180989@1b1d24b9 on org.eclipse.jetty.io.ManagedSelector@2c81ffe6 id=1 keys=1 selected=0 
2017-09-15 07:50:41,027 DEBUG o.e.j.h.HttpParser [qtp681747431-112] atEOF HttpParser{s=CLOSE,150 of 150} 
2017-09-15 07:50:41,027 DEBUG o.e.j.i.ManagedSelector [qtp681747431-31-lowPrioritySelector] Selector loop woken up from select, 0/0 selected 
2017-09-15 07:50:41,027 DEBUG o.e.j.u.t.s.ProduceExecuteConsume [qtp681747431-31-lowPrioritySelector] org.eclipse.jetty.io.ManagedSelector$SelectorProducer@2c361e6a produced org.eclipse.jetty.io.ManagedSelector$$Lambda$1418/652180989@1b1d24b9 
2017-09-15 07:50:41,027 DEBUG o.e.j.s.HttpConnection [qtp681747431-112] HttpConnection@7d66662[p=HttpParser{s=CLOSE,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null}<-SocketChannelEndPoint@63ccefa7{/10.42.136.138:52600<->/10.42.157.16:8080,CLOSED,fill=-,flush=-,to=5/30000}{io=0/0,kio=-1,kro=-1}->HttpConnection@7d66662[p=HttpParser{s=CLOSE,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null} filled -1 HeapByteBuffer@5b00e3d4[p=0,l=0,c=8192,r=0]={<<<>>>HTTP/1.1 200 OK\r\n...\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00} 
2017-09-15 07:50:41,027 DEBUG o.e.j.i.ManagedSelector [qtp681747431-31-lowPrioritySelector] Low Priority Selector executing BLOCKING org.eclipse.jetty.io.ManagedSelector$$Lambda$1418/652180989@1b1d24b9 
2017-09-15 07:50:41,027 DEBUG o.e.j.i.ManagedSelector [qtp681747431-167] Selector loop waiting on select 
2017-09-15 07:50:41,027 DEBUG o.e.j.u.t.QueuedThreadPool [qtp681747431-31-lowPrioritySelector] queue org.eclipse.jetty.io.ManagedSelector$$Lambda$1418/652180989@1b1d24b9 
2017-09-15 07:50:41,027 DEBUG o.e.j.s.HttpConnection [qtp681747431-112] HttpConnection@7d66662[p=HttpParser{s=CLOSE,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null}<-SocketChannelEndPoint@63ccefa7{/10.42.136.138:52600<->/10.42.157.16:8080,CLOSED,fill=-,flush=-,to=5/30000}{io=0/0,kio=-1,kro=-1}->HttpConnection@7d66662[p=HttpParser{s=CLOSE,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null} parse HeapByteBuffer@5b00e3d4[p=0,l=0,c=8192,r=0]={<<<>>>HTTP/1.1 200 OK\r\n...\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00} {} 
2017-09-15 07:50:41,027 DEBUG o.e.j.h.HttpParser [qtp681747431-112] parseNext s=CLOSE HeapByteBuffer@5b00e3d4[p=0,l=0,c=8192,r=0]={<<<>>>HTTP/1.1 200 OK\r\n...\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00} 
2017-09-15 07:50:41,027 DEBUG o.e.j.h.HttpParser [qtp681747431-112] CLOSE --> CLOSED 
2017-09-15 07:50:41,027 DEBUG o.e.j.u.t.QueuedThreadPool [qtp681747431-29] run org.eclipse.jetty.io.ManagedSelector$$Lambda$1418/652180989@1b1d24b9 
2017-09-15 07:50:41,027 DEBUG o.e.j.s.HttpConnection [qtp681747431-112] HttpConnection@7d66662[p=HttpParser{s=CLOSED,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null}<-SocketChannelEndPoint@63ccefa7{/10.42.136.138:52600<->/10.42.157.16:8080,CLOSED,fill=-,flush=-,to=5/30000}{io=0/0,kio=-1,kro=-1}->HttpConnection@7d66662[p=HttpParser{s=CLOSED,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null} parsed false HttpParser{s=CLOSED,150 of 150} 
2017-09-15 07:50:41,027 DEBUG o.e.j.i.ManagedSelector [qtp681747431-29] Destroyed SocketChannelEndPoint@63ccefa7{/10.42.136.138:52600<->/10.42.157.16:8080,CLOSED,fill=-,flush=-,to=5/30000}{io=0/0,kio=-1,kro=-1}->HttpConnection@7d66662[p=HttpParser{s=CLOSED,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null} 
2017-09-15 07:50:41,027 DEBUG o.e.j.s.HttpConnection [qtp681747431-112] releaseRequestBuffer HttpConnection@7d66662[p=HttpParser{s=CLOSED,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null}<-SocketChannelEndPoint@63ccefa7{/10.42.136.138:52600<->/10.42.157.16:8080,CLOSED,fill=-,flush=-,to=5/30000}{io=0/0,kio=-1,kro=-1}->HttpConnection@7d66662[p=HttpParser{s=CLOSED,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null} 
2017-09-15 07:50:41,027 DEBUG o.e.j.i.AbstractConnection [qtp681747431-29] onClose HttpConnection@7d66662[p=HttpParser{s=CLOSED,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null}<-SocketChannelEndPoint@63ccefa7{/10.42.136.138:52600<->/10.42.157.16:8080,CLOSED,fill=-,flush=-,to=5/30000}{io=0/0,kio=-1,kro=-1}->HttpConnection@7d66662[p=HttpParser{s=CLOSED,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null} 
2017-09-15 07:50:41,027 DEBUG o.e.j.u.t.QueuedThreadPool [qtp681747431-29] ran org.eclipse.jetty.io.ManagedSelector$$Lambda$1418/652180989@1b1d24b9 
2017-09-15 07:50:41,027 DEBUG o.e.j.s.HttpConnection [qtp681747431-112] HttpConnection@7d66662[p=HttpParser{s=CLOSED,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null}<-SocketChannelEndPoint@63ccefa7{/10.42.136.138:52600<->/10.42.157.16:8080,CLOSED,fill=-,flush=-,to=5/30000}{io=0/0,kio=-1,kro=-1}->HttpConnection@7d66662[p=HttpParser{s=CLOSED,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null} onFillable exit HttpChannelState@5265af91{s=IDLE a=NOT_ASYNC i=true r=!P!U w=false} null 
2017-09-15 07:50:41,027 DEBUG o.e.j.u.t.s.ExecutingExecutionStrategy [qtp681747431-112] EPC Prod/org.eclipse.jetty.io.ManagedSelector$SelectorProducer@2c361e6a invoked CEP:SocketChannelEndPoint@63ccefa7{/10.42.136.138:52600<->/10.42.157.16:8080,CLOSED,fill=-,flush=-,to=5/30000}{io=0/0,kio=-1,kro=-1}->HttpConnection@7d66662[p=HttpParser{s=CLOSED,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null}:runFillable:BLOCKING 
2017-09-15 07:50:41,028 DEBUG o.e.j.u.t.s.ExecuteProduceConsume [qtp681747431-112] EPC Prod/org.eclipse.jetty.io.ManagedSelector$SelectorProducer@2c361e6a ran CEP:SocketChannelEndPoint@63ccefa7{/10.42.136.138:52600<->/10.42.157.16:8080,CLOSED,fill=-,flush=-,to=5/30000}{io=0/0,kio=-1,kro=-1}->HttpConnection@7d66662[p=HttpParser{s=CLOSED,150 of 150},g=HttpGenerator@3e310e8{s=START}]=>HttpChannelOverHttp@7f9578fa{r=1,c=false,a=IDLE,uri=null}:runFillable:BLOCKING 



```

**Comment by lucilecoutouly** [comment]
Hello,

i need to know where i can configure the jetty server timout etc ....

thanks

**Comment by udoprog** [comment]
Jetty timeout is currently not configurable. It's currently hardcoded here:
https://github.com/spotify/heroic/blob/master/heroic-core/src/main/java/com/spotify/heroic/http/query/QueryResource.java#L170

Does your request take more than 300 seconds? This might be a hint that your data is not shaped in a way that is suitable for Heroic.

**Comment by lucilecoutouly** [comment]
thank you for your explanation but my request takes only 10 s and it breaks 
i am under 300s

what does mean 
```

Connection reset by peer
```

which is the peer  in this case ? 
which is the client ?

netty ? http ? ....

**Comment by lucilecoutouly** [comment]
Hello 

i think it was cassandra config in cassandra.yml

```yaml
counter_write_request_timeout_in_ms: 5000
range_request_timeout_in_ms: 10000
read_request_timeout_in_ms: 5000
request_timeout_in_ms: 10000
truncate_request_timeout_in_ms: 60000
write_request_timeout_in_ms: 2000
```
